src/core/featuremodel.cpp

```diff
@@ -321,7 +321,18 @@ void FeatureModel::applyGeometry()
 {
   QgsGeometry geometry = mGeometry->asQgsGeometry();
 
-  const QList<QgsVectorLayer *> intersectionLayers = QgsProject::instance()->avoidIntersectionsLayers();
+  QList<QgsVectorLayer *> intersectionLayers;
+  switch ( QgsProject::instance()->avoidIntersectionsMode() )
+  {
+    case QgsProject::AvoidIntersectionsMode::AvoidIntersectionsCurrentLayer:
+      intersectionLayers.append( mLayer );
+      break;
+    case QgsProject::AvoidIntersectionsMode::AvoidIntersectionsLayers:
+      intersectionLayers = QgsProject::instance()->avoidIntersectionsLayers();
+      break;
+    case QgsProject::AvoidIntersectionsMode::AllowIntersections:
+      break;
+  }
```

Comment:
wouldn't it be interesting to add an optional parameter to `QgsProject:: avoidIntersectionsLayers( QgsProject::AvoidIntersectionsMode mode )` so that it does the testing there?

otherwise, good!

Reply:
I was first going for that. However, I wasn't sure how one would handle passing on the current layers (maybe as an optional second parameter?). 

Anyhow, settled for more code on the app side :)

Reply:
I was first going for that. However, I wasn't sure how one would handle passing on the current layers (maybe as an optional second parameter?). 

Anyhow, settled for more code on the app side :)

Reply:
I thought it would return a list, either empty, with current layer or all.

Reply:
Yeah, but getting the active/current layer from within QgsProject alone isn't possible (unless I've missed something obvious somewhere).

Reply:
Ah correct, sorry